RateLimiter/ApiRule/RuleImplementations/MaxRequestsPerTimespanRateLimitRule.cs

```diff
@@ -0,0 +1,32 @@
+﻿using RateLimiter.Model.Enum;
+using RateLimiter.Model.Request;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace RateLimiter.ApiRule.RuleImplementations
+{
+    public class MaxRequestsPerTimespanRateLimitRule : IRuleValidation
+    {
+        protected int RequestTimespanInMinutes => 60;
+
+        protected int MaxRequestsInTimespan => 2;
+
+        public virtual bool Validate(string token, ResourceEnum resourceName, List<ApiRequest> tokenRequestLog)
```

Comment:
It is good for the future because , I would if I wished to create rules based on MaxRequestsPerTimespanRateLimitRule, I will create class which will be derived class of MaxRequestsPerTimespanRateLimitRule
 and I wiil Create  "validate" override method  because  Virtual allows an inheriting class to replace a method that the base class then uses .

Because it is not necessary to have access to other classes on this proerty . This is accessible in a derived class only